Tighten code and add magnification control to the hand panel

Baharis opened the pull request:
Introduces minor changes to the manual control panel aimed to add magnification buttons suggested by @viljarjf in #140 and tighten it a bit, both in terms of code quality/quantity and space taken in the GUI:

- "Modes" drawback now pulls modes from config instead of using hard-coded ['diff', 'mag1', 'mag2', 'lowmag', 'samag'];
- "Reset stage" button that otherwise took entire line was moved to the top of the frame, next to other buttons;
- "Stage (XY)" fields were changed from `Entry` to `Spinbox` with 100-nm increment to allow setting with mouse;
- Wobbler control changed from two mutually-exclusive buttons (Start/Stop) to a checkbutton akin defocus;
- Magnification "+" and "–" buttons added (@viljarjf );
- Removed spacing between brightness and focus controls – now the "mechanical" and "beam" controls are split neatly;
- Fixed minor typos and spacing inconsistencies, generalized some methods.

| Before | After |
| --- | --- |
| <img width="654" height="559" alt="image" src="https://github.com/user-attachments/assets/30b8ca87-b2a9-4e7b-8877-a165b83681e9" /> | <img width="655" height="560" alt="image" src="https://github.com/user-attachments/assets/000d87d6-780a-4098-805d-35c79072e43f" /> |

Note how the refreshed panel has the same height as the old one despite having a new row.

When I started editing I had large changes in mind, but I decided to scale them down and leave the spotlight on the simulator for the time being. On accident, I also found out that repeatedly turning the alpha wobbler on and off consistently crashes Instamatic. Could not fix it quickly, this is a low-priority issue for the future.

viljarjf committed:
Speed up kinematic simulations

Baharis committed:
Fix typos in Magnification error messages

Baharis committed:
Tighten handpanel, use config modes, spinbox stage xy

Baharis committed:
Remove repetition and excessive lines

Baharis committed:
Make spacing consistent

Baharis committed:
Change alpha-wobbler 2-button toggle into a checkbutton for consistency

Baharis committed:
Simulated beamshift = 0 not strictly necessary for this PR

Baharis committed:
Set columnspan=2 of wobble toggle to avoid unnecessary stretch

stefsmeets commented on src/instamatic/gui/ctrl_frame.py:
```diff
@@ -31,59 +34,47 @@ def __init__(self, parent):
-            command=self.set_mode,
-        )
-        self.o_mode.grid(row=8, column=1, sticky='W', padx=10)
+        ranges = config.microscope.ranges
+        modes = list(ranges.keys())
```
Python always iterates over the dict keys, so it's not necessary to call keys.

```suggestion
        modes = list(config.microscope.ranges)
```

Baharis replied:
The panels gets only as much love as time I can spare now, unfortunately.
Also, as `import this` verse 2 proclaimeth, *Explicit is better than implicit*

stefsmeets reviewed (approved):
Nice, love to see this little panel get some love.

viljarjf reviewed (approved):
Nice! I had some minor comments, but otherwise I think these are useful additions!

One thing I noticed reading through the file is the inconsistency of using the event queue or simply changing things directly in self.ctrl. Is that something to consider? I am not familiar with how the GUI is set up, especially regarding multiple threads/communication with servers

Baharis commented on src/instamatic/gui/ctrl_frame.py:
```diff
@@ -161,19 +152,22 @@ def __init__(self, parent):
-        frame.pack(side='top', fill='x', padx=10, pady=10)
-
-        frame = Frame(self)
+        # Magnification
+        Label(frame, text='Magnification', width=20).grid(row=14, column=0, sticky='W')
```
@viljarjf I absolutely wanted to display this value and the traces of this idea are in the code (see explicit declaration of `ranges = config.microscope.ranges` dict with this info) but once I started implementing it I noticed this requires a lot of changes, so I backpedaled for now. Maybe I'll add a print at least.

Baharis committed:
Apply changes suggested during review

Co-authored-by: Viljar Femoen <viljar.femoen@hotmail.no>

Baharis commented:
Ok, I applied suggested changes and will be merging tomorrow-ish unless anyone objects. I added print statements on magnification change, `ctrl.stage.neutral()` instead of setting zeroes, and removed unnecessary declaration. This is by no means a major definitive rework, just a small patch to help testing simulator based on @viljarjf's suggestion.

Baharis committed:
In control panel's 'Stage (XYZ)', add Z control (ugly but very useful)
